### ISSUE TITLE
Investigate load state_dict vs loading whole model

### ISSUE THREAD

**Issue by damccorm** [apache/beam#21450]
Loading pytorch model as whole has some issues with pickling. Investigate it with running some experiments. If the model size is too large, the current implementation of the RunInference for PyTorch would fail because of memory limits.

 

1. We can pass the model class to the `load_model` of PyTorchModelLoader and load the model there. This wouldn't pickle the model object but would pickle the class and the model would be instantiated on the workers.

Imported from Jira [BEAM-14368](https://issues.apache.org/jira/browse/BEAM-14368). Original Jira may contain additional context.
Reported by: Anand Inguva.
Subtask of issue #21435

**Comment by damccorm** [comment]
Unable to assign user @AnandInguva. If able, self-assign, otherwise tag @damccorm so that he can assign you. Because of GitHub's spam prevention system, your activity is required to enable assignment in this repo.

**Comment by yeandy** [comment]
Subtask of https://github.com/apache/beam/issues/22117

**Comment by damccorm** [comment]
@anandinguva is there still work to be done on this one? https://github.com/apache/beam/pull/17494 resolved this problem, right?

**Comment by AnandInguva** [comment]
No, thats a different solution on how to load the model_weights on the runner instead of SDK. that solution still needs the Model class python code.

This issue references on how to load the model without the need of model class. But as of now, there is no recommended solution in Pytorch for this. Please look at https://pytorch.org/tutorials/beginner/saving_loading_models.html#save-load-entire-model
